uikit/views/src/main/java/com/sygic/maps/uikit/views/navigation/signpost/BaseSignpostView.kt

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2019 Sygic a.s. All rights reserved.
+ *
+ * This project is licensed under the MIT License.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.sygic.maps.uikit.views.navigation.signpost
+
+import android.animation.LayoutTransition
+import android.content.Context
+import android.util.AttributeSet
+import androidx.annotation.CallSuper
+import androidx.constraintlayout.widget.ConstraintLayout
+import com.sygic.maps.uikit.views.R
+
+@Suppress("CustomViewStyleable")
+abstract class BaseSignpostView(
+    context: Context,
+    attrs: AttributeSet?,
+    defStyleAttr: Int,
+    defStyleRes: Int
+) : ConstraintLayout(context, attrs, defStyleAttr) {
+
+    private var layoutMargin: Int = 0
+    private var layoutMarginTop: Int = 0
+    private var layoutMarginBottom: Int = 0
+    private var layoutMarginStart: Int = 0
+    private var layoutMarginEnd: Int = 0
+
+    init {
+        isClickable = true
+        layoutTransition = LayoutTransition()
+
+        attrs?.let { attributeSet ->
+            context.obtainStyledAttributes(
+                attributeSet,
+                R.styleable.SignpostView,
+                defStyleAttr,
+                defStyleRes
+            ).apply {
+                setBackgroundResource(getResourceId(R.styleable.SignpostView_android_background,0))
+
+                layoutMargin = getDimensionPixelSize(R.styleable.SignpostView_android_layout_margin, 0)
+                layoutMarginTop = getDimensionPixelSize(R.styleable.SignpostView_android_layout_marginTop, 0)
+                layoutMarginBottom = getDimensionPixelSize(R.styleable.SignpostView_android_layout_marginBottom, 0)
+                layoutMarginStart = getDimensionPixelSize(R.styleable.SignpostView_android_layout_marginStart, 0)
+                layoutMarginEnd = getDimensionPixelSize(R.styleable.SignpostView_android_layout_marginEnd, 0)
```

Comment:
```suggestion
                layoutMarginEnd = it.getDimensionPixelSize(R.styleable.SignpostView_android_layout_marginEnd, 0)
```

Reply:
Nerozumím v čem je toto lepší? Přidá to jen hromadu zbytečných it-ů. Btw, jak tě to teď napadlo? Používáme to tak ve všech custom views (momentálně v 7, když nepočítám těch dalších 5 co je aktuálně na PR) a na všech jsi byl na PR, či?

Reply:
to je mozne, ze je toho viac, nemusel som si vsimnut.

Ano, su tam it-y, ale ma to aj svoj dovod. Pozri napr riadok `setBackgroundResource(getResourceId(R.styleable.SignpostView_android_background,0))`

Z tohoto zapisu netusis, ktora metoda je z `TypedArray` a ktora nie... (keby to bola napr custom trieda, ktorej metody nepoznas "z pamati", tak je z toho gulas. Compiler si s tym magicky poradi a dokonca ta nenuti volat ani `this@BaseSignpostView.setBackgroundResource(getResourceId(R.styleable.SignpostView_android_background,0))`

Moje pravidlo (ktore vychadza aj z pomenovania tych metod) je, ze `apply` pouzijem vtedy, ked chcem nastavovat (aplikovat) veci na tom objekte na ktoram volam `apply`. Tj v tomto pripade by mali byt metody `TypedArray` na lavej strane `=`. Ale ked sa pozries, ani jedna okrem `recycle` nie je na lavej strane. Vsetky su na pravej. Tj ty neaplikujes veci na tom objekte na ktorom volas `apply`, ty to pouzivas ako zdroj informacii a aplikujes to na view. Ale `apply` nie je pouzite na view.

`Also` na druhu stranu je prave to, ze tu mam **hotovy** objekt, ale chcem ho este pouzit na nejake ine veci, tak pouzijem `also` a pouzijem `it` na pravej strane `=` a vsetkym je jasne, ktore metody su koho.

Dufam, ze som to vysvetlil tak, ze sa to dalo pochopit. Viem, ze technicky to je ok, compiler to magicky rozlusit a pouzije spravne `this` na spravnom mieste, ale semanticky to spravne nie je. A semantika je podla mna dolezitejsia ako usetrenie par znakov na opakujucom sa `it`.

Reply:
souhalsím